python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -317,7 +326,11 @@ def mark_corrections(self, data, flag):
         if data.moveWhileExposing:
             self.move_while_exposing = flag
 
-        if data.all:
+        if hasattr(data, "enableAll") and data.enableAll:
```

Comment:
Surely you should know in advance whether data will have attribute "enableAll"? It it is sure to be present then it is much cleaner to just reference it: `if data.enableAll:`.

But if you really must ask, then consider `if getattr(data, "enableAll", False):` and similarly for "disableAll" a few lines below